### ISSUE TITLE
[API] Add ASCIIEncoding.IsEquivalentOrdinal(string, ROS)

### ISSUE THREAD

**Issue by benaadams** [dotnet/corefx#36519]
From https://github.com/aspnet/AspNetCore/pull/8374#discussion_r270713249

Cross encoding compare
```csharp
namespace System.Text
{
    public partial class ASCIIEncoding : Encoding
    {
        public static bool IsEquivalentOrdinal(string str, ReadOnlySpan<byte> asciiBytes);
    }
}
```

This is method to test whether a sequence of ascii bytes is the same as a string; without having to first convert the bytes to a sting and allocate and then use string equality. (e.g. compare ascii bytes to a UTF-16 string)

Used in https://github.com/aspnet/AspNetCore/pull/8374 to drop steady state string allocations to zero (as previous materialized strings can be reused when they match the on wire bytes)

Could extend with an `bool ignoreCase = false` if desired, however that is not required for this particular use case. 

/cc @GrabYourPitchforks @davidfowl

**Comment by Wraith2** [comment]
Couldn't you rent a char buffer scatter the ascii bytes into it and then compare with the existing ROS char comaprison operations?

**Comment by benaadams** [comment]
Bit slow; but that's implementation rather than api

**Comment by benaadams** [comment]
Bit slow compared to loading into vectors, widening and comparing them, also limited in concurrency by the number of ArrayPool buffers (will allocate outside that number); but that's implementation rather than api.

**Comment by svick** [comment]
Why is `str` a `string` and not a `ReadOnlySpan<char>`?

**Comment by benaadams** [comment]
> Why is `str` a `string` and not a `ReadOnlySpan<char>`?

Fair point, changed

**Comment by GrabYourPitchforks** [comment]
For API reviewers: I probably wouldn't put this on `ASCIIEncoding` for the same reason we didn't add extra static methods to `UTF8Encoding` (namely, the behavior would be surprising if somebody subclassed and customized the `Encoding` type). But there's nothing inherently wrong with the underlying scenario.

**Comment by gfoidl** [comment]
> 
> 
> > Why is `str` a `string` and not a `ReadOnlySpan<char>`?
> 
> Changed

@benaadams please also update the title to reflect this.

**Comment by GrabYourPitchforks** [comment]
Possibly covered by https://github.com/dotnet/corefx/issues/34144?

**Comment by benaadams** [comment]
Yes, think it would be covered by `Equals(ReadOnlySpan<byte> left, ReadOnlySpan<char> right)` ?
